sdv/metadata/single_table.py

```diff
@@ -245,11 +244,92 @@ def to_dict(self):
 
         return deepcopy(metadata)
 
+    def _determine_sdtype_for_numbers(self, data):
+        """Determine the sdtype for a numerical column.
+
+        Args:
+            data (pandas.Series):
+                The data to be analyzed.
+        """
+        sdtype = 'numerical'
+        if len(data) > 5:
+
+            whole_values = (data == data.round()).loc[~data.isna()].all()
+            positive_values = (data >= 0).loc[~data.isna()].all()
+
+            unique_values = data.nunique()
+            percentage_unique_values = unique_values <= round(len(data) / 10)
+
+            if whole_values and positive_values and percentage_unique_values:
+                sdtype = 'categorical'
+            elif unique_values == len(data) and whole_values:
+                sdtype = 'id'
+
+        return sdtype
+
+    def _determine_sdtype_for_objects(self, data):
+        """Determine the sdtype for an object column.
+
+        Args:
+            data (pandas.Series):
+                The data to be analyzed.
+        """
+        sdtype = None
+        try:
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=UserWarning)
+                pd.to_datetime(data)
```

Comment:
There is one more requirement towards this: All `datetimes` should be in the same `format` . My suggestion to this is:
1. Detect the datetime format.
2. Cast to `pd.to_datetime(data, datetime_fromat=detected_format, errors='raise')`
3. If error is raised then we ignore the datetimes.